src/containers/IntentService.jsx

```diff
@@ -41,6 +41,8 @@ export default class IntentService extends Component {
             .then(konnector => [konnector])
         } else if (data.dataType) {
           return this.store.findByDataType(data.dataType)
+        } else if (data.category) {
```

Comment:
I think we could avoid `else if` chaining in this whole part by doing:
```js
if (data.slug) {
  return this.store.fetchKonnectorInfos(data.slug)
    .then(konnector => [konnector]

if (data.dataType) {
  return this.store.findByDataType(data.dataType)

if (data.category) {
  return this.store.findByCategory(data.category)
}
```

What do you think ?

Reply:
What is the point ?

Reply:
So it is just me who hating `if else if` chaining ? :D

Reply:
Why do you hate it ? To me a `else if` better communicates the fact that if you pass in of the branch you will not pass in the others. In your example, it is implicit because of the `return`.